src/Session.php

```diff
@@ -0,0 +1,222 @@
+<?php
+
+declare(strict_types=1); // @codeCoverageIgnore
+
+namespace Rinq;
+
+use Rinq\Ident\SessionId;
+
+/**
+ * Session is an interface representing a "local" session, that is, a session
+ * created by a peer running in this process.
+ *
+ * Sessions are the "clients" on a Rinq network, able to issue command requests
+ * and send notifications to other sessions.
+ *
+ * Sessions are created by calling Peer::session(). The peer that creates a
+ * session is called the "owning peer".
+ *
+ * Each session has an in-memory attribute table, which can be used to store
+ * application-defined key/value pairs. A session's attribute table can be
+ * modified locally, as well as remotely by peers that have received a command
+ * request or notification from the session.
+ *
+ * The attribute table is versioned. Each revision of the attribute table is
+ * represented by the Revision interface.
+ *
+ * An optimistic-locking strategy is employed to protect the attribute table
+ * against concurrent writes. In order for a write to succeed, it must be made
+ * through a Revision value that represents the current (most recent) revision.
+ *
+ * Individual attributes in the table can be "frozen", preventing any further
+ * changes to that attribute.
+ */
+interface Session
+{
+    /**
+     * The session's unique identifier.
+     *
+     * @return SessionId The session's unique identifier.
+     */
+    public function id(): SessionId;
+
+    /**
+     * CurrentRevision returns the current revision of this session.
+     *
+     * @return Revision Current revision of this session.
+     *
+     * @throws InvalidRevisionException If the session has been closed, and revision is invalid.
+     */
+    public function currentRevision(): Revision;
+
+    /**
+     * Call sends a command request to the next available peer listening to the
+     * $namespace namespace and waits for a response.
+     *
+     * In the context of the call, the sessions owning peer is the "client" and
+     * the listening peer is the "server". The client and server may be the same
+     * peer.
+     *
+     * Both $command and $out are passed to the command handler on the server.
+     *
+     * Calls always use a deadline; if $context does not have a deadline, a
+     * timeout equal to Config::DEFAULT_TIMEOUT is used.
```

Comment:
s/deadline/timeout

I can't remember if I've come across `Config` yet, but since it's a per-peer-configuration object, it's members will need to be regular properties  - `$config->defaultTimeout` - as opposed to constants.